classes/core/Core.inc.php

```diff
@@ -551,4 +554,27 @@ private static function _getUrlComponents($urlInfo, $isPathInfo, $offset, $varNa
 
         return $component;
     }
+
+    /**
+     * Extract the class name from the given file path.
+     *
+     * @param  SplFileInfo $file info about a file extract class name from
+     * @return string fully qualified class name
+     * @see Finder
+     */
+    public static function classFromFile(SplFileInfo $file): string
+    {
+        $pathFromBase = trim(Str::replaceFirst(base_path(), '', $file->getRealPath()), DIRECTORY_SEPARATOR);
+        $libPath = 'lib' . DIRECTORY_SEPARATOR . 'pkp' . DIRECTORY_SEPARATOR;
+        $namespace = Str::startsWith($pathFromBase, $libPath) ? 'PKP\\' : 'APP\\';
+
+        $path = $pathFromBase;
+        if ($namespace === 'PKP\\') {
+            $path = Str::replaceFirst($libPath, '', $path);
+        }
+        if (Str::startsWith($path, 'classes')) {
+            $path = Str::replaceFirst('classes' . DIRECTORY_SEPARATOR, '', $path);
+        }
+        return $namespace . str_replace('/', '\\', Str::replaceLast('.inc.php', '', $path));
+    }
```

Comment:
@asmecher just a heads-up for when you do the mass file renaming. This code reads all the files in a directory and gets the fully-qualified class names. This is a common approach with Laravel, and prevents us from maintaining a manual list of these classes (events and mailables) somewhere.